packages/urban-bot/src/hooks/useText.ts

```diff
@@ -0,0 +1,24 @@
+import { UrbanBotType, UrbanEventListener } from '../types';
+import { useSubscribeWithSpreadPayload } from './hooks';
+import { matchPattern } from '../utils/matchPattern';
+
+export type Pattern = string | RegExp;
+
+export function useText<BotType extends UrbanBotType>(
+    listener: UrbanEventListener<BotType, 'text'>,
+    pattern?: Pattern | Pattern[],
+) {
+    const listenerGuard: UrbanEventListener<BotType, 'text'> = (event) => {
+        if (pattern !== undefined) {
```

Comment:
``` ts
 if (pattern !== undefined && matchPattern(event.text, pattern)) {
       listener(event);;
 }  
```

Reply:
`listener` should be always called if  `pattern` is undefined